File: java_brain/src/main/java/org/netbrains/javabrain/neural/InputNeuron.java
package org.netbrains.javabrain.neural;

public class InputNeuron extends Neuron {
    public InputNeuron(int layerIndex, int index) {
        super(layerIndex, index);
    }

    public void setValue(double value) {
        this.value = value;
    }

    @Override
    public double activate() {
        return value;
    }
}

File: cpp_brains/inputneuron.cpp
#include "inputneuron.h"

InputNeuron::InputNeuron(int layerIndex, int index)
    :Neuron{layerIndex, index}
{

}

void InputNeuron::setValue(double value)
{
    m_value = value;
}

double InputNeuron::activate()
{
    return m_value;
}
